### PR TITLE
Overflow Rerouting System

### PR DESCRIPTION
**Pull request by tigercat2000**
This commit adds an overflow rerouting system customizable by modifying
the config. 

If the config values are set, after X amount of clients have
connected, the server will start rerouting all new players to the
configuration "overflow" server.

Note that admins are immune to this, and
therefore will not be rerouted under any circumstance. 

Players that have already connected and have a body will not be rerouted either, this will
only affect new players that get sent to the lobby.

It probably would also kick players if an admin sent them to the lobby,
but that's a very special case scenario.

TL:DR; If the host configures it, the server can refuse new players after a certain amount of clients are connected, and move them to a buffer server

### CODE REVIEW

File: config/example/config.txt [hunk 1]
```diff
@@ -273,4 +273,13 @@ EVENT_DELAY_UPPER 15;45;70
...
+STARLIGHT 0
+
+## Player rerouting stuff
+## If not 0, players can be rerouted to an overflow server after a certain cap is reached
+
+ ## Cap before players start being rerouted
+PLAYER_REROUTE_CAP 0
+
+ ## Server to reroute to
+OVERFLOW_SERVER_URL byond://nanotrasen.se:6666
```
Comment (Krausus) [inline config/example/config.txt]: This should probably start commented out with an example URL, like the other config URLs.

File: code/controllers/configuration.dm [hunk 2]
```diff
@@ -81,6 +81,7 @@
 	var/forumurl = "http://baystation12.net/forums/"
 
 	var/media_base_url = "http://nanotrasen.se/media" // http://ss13.nexisonline.net/media
+	var/overflow_server_url = "byond://nanotrasen.se:6666"
```
Comment (Krausus) [inline code/controllers/configuration.dm]: Why does this have a hardcoded server-specific default value?
Why do **any** of these have a hardcoded server-specific default value?!

File: config/example/config.txt (hunk 1 above)
Comment (tigercat2000) [inline config/example/config.txt]: good point

File: code/controllers/configuration.dm (hunk 2 above)
Comment (tigercat2000) [inline code/controllers/configuration.dm]: No clue. this has a default server-specific value to ensure that if everything goes wrong, it just connects to para and doesn't break things any further.

Comment (Krausus) [inline code/controllers/configuration.dm]: Nothing should go wrong. If it's not set, it won't redirect people. Having it set to Paradise means, in a worst case scenario, someone else accidentally sets their server up to redirect to us. That's bad. Don't do that.

Comment (ZomgPonies) [inline code/controllers/configuration.dm]: Agreed. Hardcoding our IP into configuration.dm is bad m'kay.

**Comment by SamCroswell** [comment]
I approve, everyone not on the whitelist over a 60 player count should be sent to a secondary server full of nothing but Nar'Sie.
